utils/frame/sub-du/src/main.rs

```diff
@@ -0,0 +1,254 @@
+use ansi_term::{Colour::*, Style};
+use frame_metadata::{RuntimeMetadata, RuntimeMetadataPrefixed, StorageEntryType};
+use separator::Separatable;
+use structopt::StructOpt;
+use sub_storage::{get_head, get_metadata, unwrap_decoded, Hash, StorageKey};
+
+const KB: usize = 1024;
+const MB: usize = KB * KB;
+const GB: usize = MB * MB;
+
+pub const LOG_TARGET: &'static str = "sub-du";
+
+fn get_prefix(indent: usize) -> &'static str {
+	match indent {
+		1 => "├─┬",
+		2 => "│ │─┬",
+		3 => "│ │ │─",
+		_ => panic!("Unexpected indent."),
+	}
+}
+
+struct Size(usize);
+
+impl std::fmt::Display for Size {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		if self.0 <= KB {
+			write!(f, "{: <4}B", self.0)?;
+		} else if self.0 <= MB {
+			write!(f, "{: <4}K", self.0 / KB)?;
+		} else if self.0 <= GB {
+			write!(f, "{: <4}M", self.0 / MB)?;
+		}
+
+		Ok(())
+	}
+}
+
+#[derive(Debug, Clone, Default)]
+struct Module {
+	pub name: String,
+	pub size: usize,
+	pub items: Vec<Storage>,
+}
+
+impl std::fmt::Display for Module {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		let mod_style = Style::new().bold().italic().fg(Green);
+		write!(
+			f,
+			"{} {} {}\n",
+			mod_style.paint(format!("{}", Size(self.size))),
+			get_prefix(2),
+			mod_style.paint(self.name.clone())
+		)?;
+		for s in self.items.iter() {
+			write!(f, "{} {} {}\n", Size(s.size), get_prefix(3), s)?;
+		}
+		Ok(())
+	}
+}
+
+impl Module {
+	fn new(name: String) -> Self {
+		Self { name, ..Default::default() }
+	}
+}
+
+#[derive(Debug, Copy, Clone)]
+pub enum StorageItem {
```

Comment:
We have more types now.

Reply:
Double map, etc.

Reply:
Yeah, it hasn't been touched yet, WIP

Reply:
It's interesting, the `StorageEntryType` actually contains two types now `Map` and `Plain`.